### ISSUE TITLE
LSTM MSECriterion error

### ISSUE THREAD

**Issue by lakehanne** [Element-Research/rnn#243]
Hi, 

I am running into a certain bottleneck during training. This is the same dataset of issue #179 where I have a single input, six outputs. Using an LSTM set up as 

```lua
    require 'rnn'
    cost = nn.SequencerCriterion(nn.MSECriterion())
    neunet = nn.Sequential()
    local inputSize = opt.hiddenSize
    for i,hiddenSize in ipairs(opt.hiddenSize) do 
      local rnn = nn.Sequencer(nn.LSTM(ninputs, hiddenSize, opt.rho)) 
      neunet:add(rnn)        
      if opt.dropout then
        neunet:add(nn.Sequencer(nn.Dropout(opt.dropoutProb)))
      end
      if opt.dropout then
        neunet:insert(nn.Dropout(opt.dropoutProb), 1)
      end
       inputSize = hiddenSize
    end
    -- output layer
    neunet:add(nn.Sequencer(nn.Linear(inputSize, 1)))
    neunet:add(nn.Sequencer(nn.ReLU()))
    neunet = nn.Repeater(neunet, noutputs)
```

I train as follows 

```lua
  epoch = epoch or 1

  --do one epoch
  offsets = {}
  --form mini batch    
  local iter = 1
  local target, input = {}, {}      
    for i=1,args.batchSize do
       table.insert(offsets, math.ceil(math.random()*args.batchSize))
    end
    offsets = torch.LongTensor(offsets)
    offsets = transfer_data(offsets)

     -- 1. create a sequence of rho time-steps
    local inputs, targets = {}, {}
    for step = 1, args.rho do
      inputs[step] = train_input:index(1, offsets) 
      --increase offsets indices by 1      
      offsets:add(1) -- increase indices by 1
      offsets[offsets:gt(height)] = 1
      offsets = transfer_data(offsets)  

      targets[step] = {train_out[1]:index(1, offsets), train_out[2]:index(1, offsets), 
                        train_out[3]:index(1, offsets), train_out[4]:index(1, offsets), 
                        train_out[5]:index(1, offsets), train_out[6]:index(1, offsets)}
      offsets = torch.LongTensor():resize(offsets:size()[1]):copy(offsets)
    end
     neunet:zeroGradParameters()
    inputs_, outputs = {}, {}
    local loss = 0
    outputs = neunet:forward(inputs)
    loss = cost:forward(outputs, targets)
    print(string.format("Step %d, Loss = %f ", iter, loss))
    if iter % 10 == 0 then collectgarbage() end 
    local  gradOutputs = cost:backward(outputs, targets)
    local gradInputs  = neunet:backward(inputs, gradOutputs) 
    neunet:updateParameters(opt.rnnlearningRate)
    iter = iter + 1 
```
I get errors in the cost function 

```bash
/home/lex/torch/install/share/lua/5.1/nn/MSECriterion.lua:13: attempt to call field 'new' (a nil value)
```

which is from the line 
```lua   
loss = cost:forward(outputs, targets) 
```
 in my code and it seems to suggest that the cost function is not finding the inputs to calculate the forward loss. 

What could I be doing wrong?

**Comment by lakehanne** [comment]
Never mind. Fixed. I did not arrange my inputs properly. One more thing, `nn.Sequential` was not expecting a `recursed` module in each of the layers as I had above. Intead, I designed my LSTM as follows:

```lua
    require 'rnn'
    require 'nngraph'
    -- opt.hiddenSize = loadstring(" return "..opt.hiddenSize)()
    nn.LSTM.usenngraph = true -- faster
    --cost = nn.SequencerCriterion(nn.DistKLDivCriterion())
    local crit = nn.MSECriterion()
    cost = nn.SequencerCriterion(crit)
    neunet = nn.Sequential()
    local inputSize = opt.hiddenSize[1]
    for i, inputSize in ipairs(opt.hiddenSize) do 
      local rnn = nn.LSTM(ninputs, opt.hiddenSize[1], opt.rho)
      neunet:add(rnn) 
       
      if opt.dropout then
        neunet:insert(nn.Dropout(opt.dropoutProb), 1)
      end
       inputSize = opt.hiddenSize[1]
    end

    -- output layer
    neunet:add(nn.Linear(ninputs, 1))
    neunet:add(nn.ReLU())

    -- will recurse a single continuous sequence
    neunet:remember('eval')
    --output layer 
    neunet = nn.Repeater(neunet, noutputs)
```

Closing.
